src/AppInstallerCLICore/Workflows/ArchiveFlow.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "ArchiveFlow.h"
+#include "winget/Archive.h"
+
+namespace AppInstaller::CLI::Workflow
+{
+    void ExtractInstallerFromArchive(Execution::Context& context)
+    {
+        const auto& installerPath = context.Get<Execution::Data::InstallerPath>();
+        const auto& installerParentPath = installerPath.parent_path();
+
+        HRESULT hr = AppInstaller::Archive::ExtractArchive(installerPath, installerParentPath);
+
+        if (SUCCEEDED(hr))
+        {
+            AICLI_LOG(CLI, Info, << "Successfully extracted archive");
+            context.SetFlags(Execution::ContextFlag::InstallerExtractedFromArchive);
+        }
+        else
+        {
+            AICLI_LOG(CLI, Info, << "Failed to extract archive");
```

Comment:
Can we also log the path where it failed to extract to?

Reply:
added path to log